src/App.tsx

```diff
@@ -1,26 +1,163 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import { UserWarning } from './UserWarning';
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
 
-const USER_ID = 0;
+import { Header } from './commponents/Header';
+import { Footer } from './commponents/Footer';
+import { Errors } from './commponents/Errors';
+import { TodoCard } from './commponents/TodoCard';
+
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { ErrorType } from './types/Errors';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<Status>(Status.All);
+  const [errorType, setErrorType] = useState<string>('');
+  const [newTodo, setNewTodo] = useState<Todo | null>(null);
+  const [deleteItemTodo, setDeleteItemTodo] = useState(NaN);
+  const [loadedDelete, setLoadedDelete] = useState(false);
+  const [todoTask, setTodoTask] = useState('');
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      const timeoutId = setTimeout(() => setErrorType(''), 3000);
+
+      try {
+        const todosData = await getTodos();
+
+        setTodos(todosData);
+      } catch {
+        setErrorType(ErrorType.UnableToLoad);
+        clearTimeout(timeoutId);
+      }
+
+      return () => clearTimeout(timeoutId);
+    };
+
+    fetchTodos();
+  }, []);
+
+  const countTodo = todos.filter(todo => !todo.completed).length;
+
+  const todoFilter = todos.filter(todo => {
+    if (filterType === Status.Active) {
+      return !todo.completed;
+    }
+
+    if (filterType === Status.Completed) {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const addNewTodo = useCallback(async (todoToAdd: Todo) => {
+    setNewTodo(todoToAdd);
+
+    try {
+      const todoNew = await addTodo(todoToAdd);
+
+      setTodos(currentTodos => [...currentTodos, todoNew]);
+      setTodoTask('');
+    } catch {
+      setErrorType(ErrorType.UnableToAdd);
+    } finally {
+      setNewTodo(null);
+    }
+  }, []);
+
+  const deleteTodoItem = useCallback(async (todoId: number) => {
+    setDeleteItemTodo(todoId);
+
+    try {
+      await deleteTodo(todoId);
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    } catch {
+      setErrorType(ErrorType.UnableToDelete);
+    } finally {
+      setDeleteItemTodo(NaN);
+    }
```

Comment:
Using `useCallback` is useless here
We should use it when we also use `React.memo` - https://react.dev/reference/react/memo